### PR TITLE
feat: implement local Telemetry suite

### LINKED ISSUE #3301: Easy Setup for Telemetry Tools Suite for Local Bacalhau Usage
## Description

Developers need a straightforward way to set up a comprehensive telemetry suite for inspecting telemetry data emitted from Bacalhau. This suite should facilitate the monitoring and analysis of metrics and traces generated by Bacalhau, enhancing the observability and debugging capabilities for developers.

## Proposed Telemetry Suite Components

The telemetry suite should include the following components:

- **Prometheus Instance**: For storing and managing metrics data.
- **Grafana Instance**: To provide a visualization platform for analyzing metrics.
- **Jaeger Instance**: For detailed inspection and visualization of tracing data.
- **OpenTelemetry Collector**: To export metrics emitted by the Bacalhau OpenTelemetry (OTEL) SDK.
- **Documentation/Guidelines**: Detailed instructions on how to set up and utilize the telemetry suite.

### PR DESCRIPTION
**Pull request by frrist**
- closes #3301

### CODE REVIEW

**Review by rossjones — APPROVED** [review]
Looks great! Tried it out and all seemed to work.  I think we need to find a way to configure the prom data source for grafana so we can see any metrics published, but the jaeger stuff seems to work out of the box! 

Might want to link https://github.com/bacalhau-project/bacalhau/blob/main/docs/docs/dev/open_telemetry_in_bacalhau.md to this work.

Also could update https://github.com/bacalhau-project/bacalhau/blob/main/docs/docs/running-node/observability.md to point to this to make it easier.

**Comment by frrist** [comment]
> Looks great! Tried it out and all seemed to work. I think we need to find a way to configure the prom data source for grafana so we can see any metrics published, but the jaeger stuff seems to work out of the box!

Addressed in latest push, now ships with a default dashboard and Prometheus data-source automatically configured. Should just work out of the box after login.

> Might want to link https://github.com/bacalhau-project/bacalhau/blob/main/docs/docs/dev/open_telemetry_in_bacalhau.md to this work.
Also could update https://github.com/bacalhau-project/bacalhau/blob/main/docs/docs/running-node/observability.md to point to this to make it easier

Good idea, will do.

> I'm not sure if it is related to this PR, or whether it happens any time that otel is set up, but I was seeing the following in my console when trying it out.

Yeah, I think we are putting some bad data in a trace somewhere, follow on to this work is debugging that with this.

**Comment by frrist** [comment]
Not sure why docs CI is failing. Link appears to be correct when inspecting locally...

File: docs/docs/running-node/observability.md [hunk 1]
```diff
@@ -26,84 +26,5 @@ Traces are produced for all major pieces of work when processing a job, although
...
-```yaml title="prometheus.yaml"
-scrape_configs:
-  - job_name: 'otel-collector'
-    scrape_interval: 10s
-    static_configs:
-      - targets: ['otel-collector:8889']
-      - targets: ['otel-collector:8888']
-```
+To view the data locally, or simply to not use a SaaS offering, you can start follow the directions 
+[here](../../../ops/metrics/README.md)
```
Comment (rossjones) [inline docs/docs/running-node/observability.md:30]: Does this need to be a github link rather than relative to docs?

Comment (frrist) [reply docs/docs/running-node/observability.md:30]: I can't link to github because this file doesn't exist there yet, I just added it in this PR.

Comment (frrist) [reply docs/docs/running-node/observability.md]: I can't link it to github becasue its a doc I added in this PR

**Review by rossjones — APPROVED** [review]

**Pull request merged**